File: backend/src/main/java/ma/zs/alc/service/facade/collaborator/course/ParcoursCollaboratorService.java
package ma.zs.alc.service.facade.collaborator.course;

import ma.zs.alc.bean.core.course.Parcours;
import ma.zs.alc.dao.criteria.core.course.ParcoursCriteria;
import ma.zs.alc.zynerator.service.IService;

import java.util.List;


public interface ParcoursCollaboratorService extends IService<Parcours, ParcoursCriteria> {

    List<Parcours> findByCollaboratorId(Long id);

    int deleteByCollaboratorId(Long id);

    long countByCollaboratorId(Long id);

    List<Parcours> findByForExgFlexy();

    List<Parcours> findForCurrentCollaborator(String username);


    List<Parcours> findByLibelleLikeForEngFlexy(String libelle);
    List<Parcours> findByLibelleLikeForCurrent(String libelle);
}

File: frontend/src/app/shared/service/collaborator/course/ParcoursCollaborator.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {ParcoursDto} from '../../../model/course/Parcours.model';
import {ParcoursCriteria} from '../../../criteria/course/ParcoursCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";
import {Observable} from "rxjs";
import {AuthService} from "../../../../zynerator/security/shared/service/Auth.service";


@Injectable({
    providedIn: 'root'
})
export class ParcoursCollaboratorService extends AbstractService<ParcoursDto, ParcoursCriteria> {
    constructor(private http: HttpClient, private auth:AuthService) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'collaborator/parcours/';
    }

    public constrcutDto(): ParcoursDto {
        return new ParcoursDto();
    }

    public findForCurrentCollaborator(): Observable<ParcoursDto[]> {
        return this.http.get<ParcoursDto[]>(`${this.API}current/id/${this.auth.authenticatedUser.id}`);
    }

    public findByForExgFlexy(): Observable<ParcoursDto[]> {
        return this.http.get<ParcoursDto[]>(`${this.API}exgflexy`);
    }

    constrcutCriteria(): ParcoursCriteria {
        return new ParcoursCriteria();
    }


    findByLibelleLikeForEngFlexy(libelle: string): Observable<ParcoursDto[]> {
    return this.http.get<ParcoursDto[]>(`${this.API}engflexy/libelle/${libelle}`);
}

    findByLibelleLikeForCurrent(libelle: string): Observable<ParcoursDto[]> {
    return this.http.get<ParcoursDto[]>(`${this.API}current/libelle/${libelle}`);
}
}
